guitar-dongle/src/main.cpp

```diff
@@ -0,0 +1,134 @@
+#include <Arduino.h>
+#include <WiFi.h>
+#include <unordered_map>
+#include <esp_now.h>
+#include "esp_wifi.h" 
+#include "midi_common.h"
+
+#define MIDI_UART_RX_PIN 16
+
+const unsigned long WIRELESS_TIMEOUT_MILLIS = 1500;               //If wee don't receive wireless (esp-now) ping from the device in 1.5s we note off currently on'ed notes.
+const unsigned long WIRELESS_IGNORE_AFTER_LAST_UART_WRITE = 3000; //We will ignore wireless (esp-now) messages if we received direct MIDI messages in last 3s
+
+enum Protocol {
+  UART,
+  ESP_NOW,
+};
+
+typedef struct state {
+  uint8_t value;
+  Protocol protocol;
+  unsigned long changedAt;
+} state;
+
+unsigned long now = 0;
+unsigned long lastGuitarWirelessPing = 0;
```

Comment:
maybe it would make sense to have a map<channel, ChannelState> instead of duplicating everything for guitar/bass.

ChannelState may include last write time, ping, note states etc.